### PR TITLE
[DependencyInjection] Add `closure` argument type

### LINKED ISSUE #16667: [DependencyInjection] Add argument type `closure` to help passing closu…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#45878
| PR author(s) | @nicolas-grekas
| Merged in    | 6.1

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fixes https://github.com/symfony/symfony-docs/issues/16667

### CODE REVIEW

File: service_container.rst [hunk 1]
```diff
@@ -700,6 +700,100 @@ For a full list of *all* possible services in the container, run:
...
+
+    class MessageHashGenerator
+    {
+        public function __invoke(): string
+        {
+            // Compute and return a message hash
+        }
+    }
+
+Now, our configuration would look like this:
```
Comment (alamirault) [inline service_container.rst]: We can maybe avoid repeat `Now, ...` which is used just above

Comment (alexandre-daubois) [reply service_container.rst]: You're right, updated!

**Review by alamirault — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thanks Alex! Yet another superb contribution 🙇 I'm running out of messages to thank you for your amazing contributions 😃 

Note: I merged it in 6.2 and up because 6.0 and 6.1 are no longer supported.

**Comment by alexandre-daubois** [comment]
Thank you very much Javier! 🙏😃

**Pull request merged**